pkg/etcd/s3.go

```diff
@@ -32,6 +34,9 @@ type S3 struct {
 // copy of the config.Control pointer and initializes
 // a new Minio client.
 func NewS3(ctx context.Context, config *config.Control) (*S3, error) {
+	if config.EtcdS3BucketName == "" {
```

Comment:
This will cause the call below to create the client to fail. But if we want to add this, it should probably be added in pkg/cli/server where it's assigned to the config struct to short circuit it sooner. As well as the etcd-snapshot subcommands too.

Reply:
Yes, that was intentional because this is utilized in the `initS3IfNil`, and repeated use of it when attempting to save an S3 snapshot caused redundant error messages and unnecessary exercising of the rest of the code below. 

I'm a fan of performing validation at the "root" rather than on the branches (i.e. at the various pkg/cli commands) to work with DRY philosophy